runtime/compiler/control/J9Options.cpp

```diff
@@ -2474,6 +2786,331 @@ J9::Options::printPID()
+   appendRegex(options->_memUsage, curPos);
+   appendRegex(options->_classesWithFolableFinalFields, curPos);
+   appendRegex(options->_disabledIdiomPatterns, curPos);
+   options->_osVersionString = NULL;
+   options->_logListForOtherCompThreads = NULL;
+   options->_objectFileName = NULL;
+
+   // Append the data pointed by a pointer to the content and patch the pointer
+   // as a self-referring-pointer, or a relative pointer, which is
+   // the offset of the data with respect to the pointer.
+   curPos = appendContent(options->_logFileName, curPos, logFileNameLength);
+   curPos = appendContent(options->_suffixLogsFormat, curPos, suffixLogsFormatLength);
+   curPos = appendContent(options->_blockShufflingSequence, curPos, blockShufflingSequenceLength);
+   curPos = appendContent(options->_induceOSR, curPos, induceOSRLength);
+
+   // Send rtResolve option to the server:
+   // Temporary solution until we can send jitConfig->runtimeFlags to the server
+   // or make rtResolve part of TR::Options instead of runtime flags
+   auto *jitConfig = (J9JITConfig *) _feBase;
+   bool rtResolve = jitConfig->runtimeFlags & J9JIT_RUNTIME_RESOLVE;
+   char *rtResolveStr = (char *) &rtResolve;
+   curPos = appendContent(rtResolveStr, curPos, sizeof(bool));
+
+   return optionsStr;
+   }
+
+TR::Options *
+J9::Options::unpackOptions(char *clientOptions, size_t clientOptionsSize, TR::CompilationInfoPerThreadBase* compInfoPT, TR_J9VMBase *fe, TR_Memory *trMemory)
+   {
+   TR::Options *options = (TR::Options *)trMemory->allocateHeapMemory(clientOptionsSize);
+   memcpy(options, clientOptions, clientOptionsSize);
+
+   // Convert relative pointers to absolute pointers
+   // pointer = address of field + offset
+   if (options->_logFileName)
+      options->_logFileName = (char *)((uint8_t *)&(options->_logFileName) + (ptrdiff_t)options->_logFileName);
+   if (options->_suffixLogsFormat)
+      options->_suffixLogsFormat = (char *)((uint8_t *)&(options->_suffixLogsFormat) + (ptrdiff_t)options->_suffixLogsFormat);
+   if (options->_blockShufflingSequence)
+      options->_blockShufflingSequence = (char *)((uint8_t *)&(options->_blockShufflingSequence) + (ptrdiff_t)options->_blockShufflingSequence);
+   if (options->_induceOSR)
+      options->_induceOSR = (char *)((uint8_t *)&(options->_induceOSR) + (ptrdiff_t)options->_induceOSR);
+
+   // Receive rtResolve: J9JIT_RUNTIME_RESOLVE
+   // NOTE: This relies on rtResolve being the last option in clientOptions
+   // the J9JIT_RUNTIME_RESOLVE flag from JITServer client
+   // On JITServer, we store this value for each client in ClientSessionData
+   bool rtResolve = (bool) *((uint8_t *) options + clientOptionsSize - sizeof(bool));
+   // JITSERVER_TODO guards the code that relies on other JITServer unmerged files.
+#if defined(JITSERVER_TODO)
+   compInfoPT->getClientData()->setRtResolve(rtResolve);
+   _reportByteCodeInfoAtCatchBlock = fe->getReportByteCodeInfoAtCatchBlock();
+#endif /* defined(JITSERVER_TODO) */
```

Comment:
You have to unpack `_traceForCodeMining` right?

Reply:
Yes, I believe so.